x-pack/plugins/observability_solution/infra/public/components/logs_deprecation_callout.tsx

```diff
@@ -44,14 +48,22 @@ interface LogsDeprecationCalloutProps {
 
 export const LogsDeprecationCallout = ({ page }: LogsDeprecationCalloutProps) => {
   const {
-    services: { share },
+    services: {
+      share,
+      application: {
+        capabilities: { discover, fleet },
+      },
+    },
   } = useKibanaContextForPlugin();
 
   const { dismissalStorageKey, message } = pageConfigurations[page];
 
   const [isDismissed, setDismissed] = useLocalStorage(dismissalStorageKey, false);
 
-  if (isDismissed) {
+  const allDatasetLocator =
+    share.url.locators.get<AllDatasetsLocatorParams>(ALL_DATASETS_LOCATOR_ID);
+
+  if (isDismissed || !(allDatasetLocator && discover?.show && fleet?.read)) {
```

Comment:
Would be great if here too `isLogsExplorerAppAccessible` is consumed instead of `discover?.show && fleet?.read`.